### PR TITLE
Optimize context retrieving

### PR DESCRIPTION
**Pull request by Neylix**
# Description

On mainnet we can see that some cross validation nodes do not have the time to send the mining context to the coordinator before the 3ms of timeout.
This can be explained due to high latency but also the way a node retrieve the context.
Currently a node fetch the previous transaction and the p2p view concurrently and then fetch the utxos.
This is improved by fetching the utxos concurrently with other fetch.

Also fixed the timestamp of transaction fetching

# Checklist:

- My code follows the style guidelines of this project
- I have performed a self-review of my own code
- I have commented my code, particularly in hard-to-understand areas
- I have made corresponding changes to the documentation
- My changes generate no new warnings
- I have added tests that prove my fix is effective or that my feature works
- New and existing unit tests pass locally with my changes
- Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

**Review by bchamagne — APPROVED** [review]

File: lib/archethic/mining/transaction_context.ex [hunk 1]
```diff
@@ -58,9 +58,11 @@ defmodule Archethic.Mining.TransactionContext do
       |> Enum.uniq()
 
     prev_tx_task = request_previous_tx(previous_address, prev_tx_nodes_split)
+    utxos_task = request_utxos(previous_address, unspent_outputs_nodes_split)
     nodes_view_task = request_nodes_view(node_public_keys)
 
     prev_tx = Task.await(prev_tx_task)
+    utxos = Task.await(utxos_task)
```
Comment (samuelmanzanera) [inline lib/archethic/mining/transaction_context.ex:65]: You could probably use [Task.await_many](https://hexdocs.pm/elixir/Task.html#await_many/2)

Comment (Neylix) [reply lib/archethic/mining/transaction_context.ex]: Done !

File: lib/archethic/mining/transaction_context.ex [hunk 2]
```diff
@@ -98,23 +91,26 @@ defmodule Archethic.Mining.TransactionContext do
...
-
-          {:error, _} ->
-            nil
-        end
+    Task.Supervisor.async(TaskSupervisor, fn ->
+      # Timeout of 2.5 sec because the coordinator node wait 3 sec to get the context
+      # from the cross validation nodes
+      case TransactionChain.fetch_transaction(previous_address, nodes,
+             search_mode: :remote,
+             timeout: 2500
```
Comment (samuelmanzanera) [inline lib/archethic/mining/transaction_context.ex:99]: I'm wondering if it should be based on the variable of the coordinator waiting time to be more dynamic.

Comment (Neylix) [reply lib/archethic/mining/transaction_context.ex]: Deleted the commit, we'll need to rework globally on mining timeout

**Review by samuelmanzanera — APPROVED** [review]

**Pull request merged**